openff/toolkit/utils/exceptions.py

```diff
@@ -71,45 +68,218 @@ class LicenseError(ToolkitUnavailableException):
     """This function requires a license that cannot be found."""
 
 
-class InvalidToolkitError(MessageException):
+class InvalidToolkitError(OpenFFToolkitException):
     """A non-toolkit object was received when a toolkit object was expected"""
 
 
-class InvalidToolkitRegistryError(MessageException):
+class InvalidToolkitRegistryError(OpenFFToolkitException):
     """An object other than a ToolkitRegistry or toolkit wrapper was received"""
 
 
-class UndefinedStereochemistryError(MessageException):
+class UndefinedStereochemistryError(OpenFFToolkitException):
     """A molecule was attempted to be loaded with undefined stereochemistry"""
 
 
 class GAFFAtomTypeWarning(RuntimeWarning):
     """A warning raised if a loaded mol2 file possibly uses GAFF atom types."""
 
 
-class ChargeMethodUnavailableError(MessageException):
+class ChargeMethodUnavailableError(OpenFFToolkitException):
     """A toolkit does not support the requested partial_charge_method combination"""
 
 
-class IncorrectNumConformersError(MessageException):
+class IncorrectNumConformersError(OpenFFToolkitException):
     """The requested partial_charge_method expects a different number of conformers than was provided"""
 
 
 class IncorrectNumConformersWarning(Warning):
     """The requested partial_charge_method expects a different number of conformers than was provided"""
 
 
-class ChargeCalculationError(MessageException):
+class ChargeCalculationError(OpenFFToolkitException):
     """An unhandled error occured in an external toolkit during charge calculation"""
 
 
-class InvalidIUPACNameError(MessageException):
+class InvalidIUPACNameError(OpenFFToolkitException):
     """Failed to parse IUPAC name"""
 
 
-class AntechamberNotFoundError(MessageException):
+class AntechamberNotFoundError(OpenFFToolkitException):
     """The antechamber executable was not found"""
 
 
-class ParseError(MessageException, ValueError):
+class SMILESParseError(ValueError):
     """The record couple not be parsed into the given format"""
+
+
+class InvalidConformerError(Exception):
+    """
+    This error is raised when the conformer added to the molecule
+    has a different connectivity to that already defined.
+    or anyother conformer related issues.
+    """
+
+
+class SmilesParsingError(Exception):
+    """
+    This error is rasied when parsing a smiles string results in an error.
+    """
```

Comment:
> It turns out there are two ParseError in the code! One is for failing to parse SMIRNOFF source, the other is for when a wrapped toolkit fails to parse a SMILES string. I've renamed each to be more descriptive, deprecated the SMIRNOFF one in a way that shouldn't break existing code, and removed the SMILES one as I think it's less likely that somebody is capturing that specific exception.

Were they the original `SMILESParseError` and `SmilesParsingError`? I still don't understand the difference even with looking at the docstrings -- I put my understanding of the errors below. The similar names are very confusing.

```suggestion
class SMILESParseError(ValueError):
    """The record could not be parsed into the given format"""


class InvalidConformerError(Exception):
    """
    This error is raised for any conformer related issues.
    """


class SmilesParsingError(Exception):
    """
    This error is raised when a toolkit package cannot parse a SMILES string.
    """
```

Reply:
This is just me being sloppy. They should be the same thing, and also inherit as you describe